dttools/src/libforce_halt_enospc.c

```diff
@@ -65,3 +46,27 @@ int open(const char *path, int flags, ...)
 
 	return fd;
 }
+
+ssize_t write(int fd, const void *buf, size_t count) {
+
+	__typeof__(write) *original_write = dlsym(RTLD_NEXT, "write");
+
+	int real_count;
+	int prev_errno = errno;
+	errno = 0;
+	real_count = original_write(fd, buf, count);
+
+	if(real_count < 0 && errno == ENOSPC) {
+		int fd;
+		fd = open("../disk_alloc_report.txt", O_RDWR | O_CREAT);
```

Comment:
Aha, see my comment below. Perhaps we can make the name not to be fixed, which will help with relative paths.
One option is to always create the files as temps in /tmp, and communicate the name to your library somehow.